src/wallet/coin-selector.ts

```diff
@@ -36,20 +36,39 @@ export class CoinSelector {
 
         const selected = this.selectCoins(pointers, target);
 
-        const change =
+        let change =
             selected.reduce(
                 (acc, index) => acc + pointers[index].effectiveValue,
                 0,
             ) - target;
 
-        // TODO: check if change is lower than dust limit, if so add it to fee
+        // Calculate the cost of change
+        const costOfChange = this.costOfChange();
+
+        // Check if change is less than the cost of change
+        if (change <= costOfChange) {
+            change = 0;
+        }
 
         return {
             coins: selected.map((index) => pointers[index].coin),
             change,
         };
     }
 
+    costOfChange() {
```

Comment:
let's take advantage of TS syntactic sugar
```suggestion
    get costOfChange() {
```

FYI after you do this, you'd need to update the usages as well